Python server/develop 기능 정리

ssooda opened the pull request:
게시판 목록
게시글 목록
게시글 작성, 수정, 조회, 삭제
댓글 작성, 조회, 삭제
대댓글 작성, 삭제

기능 정리
컨벤션 정리

ssooda committed:
insert_board_list

board_list db에 직접 insert

ssooda committed:
board 추가, 삭제 api

내부에서만 요청하는 api
추후 board_list 업데이트 고려하여 별도 api로 만들었습니다.

ssooda committed:
insert_board_list 함수 삭제

insert_board_list 함수 삭제

ssooda committed:
댓글 관련 기능 구현 틀

ISSUES
1. 보통 댓글, 대댓글은 수정기능은 없는듯?
2. 함수명, CRUD 불일치 문제 (불일치해도 Postman은 작동하긴 함....)

http://0.0.0.0:5000/comment

1️⃣댓글 작성 (method: POST)
- query string x
- header x
- body
{
                "board_type": "secret",
                "content_id": string,
                "comment": {
                    "author": string,
                    "content": string,
                    "date": string? date?,
                    "likes": integer
                },
                "subcomment": [
                    {
                        "author": string,
                        "content": string,
                        "date": string?,
                        "likes": integer
                    },
                    {
                         "author": string,
                         "content": string,
                         "date": string? date?,
                        "likes": integer
                    }
                ]
  }

2️⃣댓글 조회 (method:GET)
-query string : content-id(게시글 고유 id), board-type
-header x
-body x

3️⃣댓글 수정 (method:PUT)
-query string x
-haeder x
-body O

4️⃣댓글 삭제
-query string x
-header x
- body : board_type, comment _id (댓글 고유id)


http://0.0.0.0:5000/subcomment

1️⃣대댓글 작성 (method: POST)
{
    "board_type": "secret",
    "comment_id": string,
    "subcomment":
    {
        "author": string,
        "content": string,
        "date": "2:50",
        "likes": integer
    }
}

2️⃣대댓글 삭제 (method:DELETE)
-query string x
-header x
-body 
{
    "board_type": "secret",
    "comment_id": string,
    "subcomment":
    {
        "author": string,
        "content": string,
        "date": "2:50",
        "likes": integer
    }
}

ssooda committed:
댓글삭제 기능 보완

1. 대댓글이 있는 상태에서 삭제
2. 대댓글이 없는 상태에서 삭제

YouminYoumin committed:
comment.py 내용 post.py에 이관

YouminYoumin committed:
Get 호출 시 viewCount 업데이트, 게시글 수정 시 article activity 분리하여 수정

ssooda committed:
api_helper와 호응, pymongo 새로운 기능 추가

ssooda committed:
기능 정리, 추가

1. post.py
- 댓글조회
- 인기게시판에 추가 (삭제 기능은 아직)
- 변수명 맞추기
- date 시도 (확실 x)

2. board.py
- newPost 시도
- 인기게시판 postlist불러오는 것 구현

3. user.py
- activity 이관

ssooda committed:
인기게시판, 작성자 탈퇴 시, 참조고려

1. post.py
- 인기게시판에 추가, 제외 기능 보완
- 작성자 탈퇴 여부 확인 후 author 값 변경 ** author 데이터 구조에 대한 논의 필요 (id, nickname, url ?)
* 댓글 조회 분리 여부 논의 필요 ( 댓글 작성자 author 관련)

2. user.py
참조 고려하여 
"_id"값을 사용자 id로

ssooda committed:
게시판 리스트, 게시글 리스트, 게시글, 댓글, 대댓글

기능 정리

ssooda committed:
Delete post.py

ssooda committed:
Delete user.py

ssooda committed:
queryStatus 수정

ssooda committed:
Delete activity.py

YouminYoumin reviewed (approved):
기능 구현때문에 고생이 많습니다 ㅎㅎ 수고하셨어요!

ssooda committed:
post.py 리펙토링 + 최신글 불러오는 api 구현

1. post.py 겹치는 부분 함수로 뺌

2. 최신글 리스트 불러오는 api 구현

3. 게시글 리스트 불러오는 uri query 변경
lastPostId => standardId
pageSize => volume
direction 추가
